### PR TITLE
RISC-V support

### PR DESCRIPTION
**Pull request by ilya-epifanov**
Original PR: https://github.com/ferrous-systems/async-on-embedded/pull/3

### CODE REVIEW

**Comment by ilya-epifanov** [comment]
@Disasm I had to rebase it onto the latest master

File: async-embedded/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+#[cfg(all(any(target_arch = "riscv32", target_arch = "riscv64"), feature = "riscv-wait-wfi-single-hart"))]
+#[inline]
+/// Wait for an interrupt or until notified by other hart via `signal_task_ready`
+/// This particular implementation decides whether to sleep or not by checking
+/// a global mutable flag that's set by `signal_task_ready`
+pub(crate) unsafe fn wait_for_event() {
+    if !TASK_READY {
+        riscv::asm::wfi();
+        TASK_READY = false;
+    }
```
Comment (Disasm) [inline async-embedded/src/lib.rs]:
```suggestion
    }
    TASK_READY = false;
```

Oops, I found a mistake

File: async-embedded/Cargo.toml [hunk 2]
```diff
@@ -2,14 +2,25 @@
...
+
+[target.'cfg(target_arch = "arm")'.dependencies]
+cortex-m = "0.6.2"
+cortex-m-udf = { path = "../cortex-m-udf" }
+
+[target.'cfg(any(target_arch = "riscv32", target_arch = "riscv64"))'.dependencies]
+riscv = "0.5.6"
+
+[features]
+default = ["riscv-wait-nop"]
```
Comment (Disasm) [inline async-embedded/Cargo.toml:23]: I'm not sure if it's a good idea to leave this default here, as user will have to pass `no-default-features = true` each time they want a different option.

File: async-embedded/src/lib.rs (hunk 1 above)
Comment (ilya-epifanov) [reply async-embedded/src/lib.rs]: The "riscv-wait-wfi-single-hart" impl already relies on the fact that you only `wait_for_event()/signal_event_ready()` from thread context. So, I don't see how `TASK_READY` could be set to `true` during `wait_for_event()`.

File: async-embedded/Cargo.toml (hunk 2 above)
Comment (ilya-epifanov) [reply async-embedded/Cargo.toml:23]: Yeah, I miss the "enum-style" features so much. But this is a more or less standard way nowadays, to preselect some working feature as a default.

File: async-embedded/src/lib.rs (hunk 1 above)
Comment (Disasm) [reply async-embedded/src/lib.rs]: Currently, `TASK_READY = true` sticks if you do not have `wait_for_event()` running "at the same time". Once set, there is not way to reset it back.

File: async-embedded/Cargo.toml (hunk 2 above)
Comment (Disasm) [reply async-embedded/Cargo.toml:23]: Ok, I do not have strong feelings about this anyway

**Review by Disasm — APPROVED** [review]

**Pull request merged**